app.rb

```diff
@@ -119,6 +119,30 @@ def project_path
       redirect to("/")
     end
 
+    post "/files" do
+      upload(project_path, params['file'])
+    end
+
+    post "/files/*" do
+      path = "#{project_path}/#{params['splat'].join('/')}"
+      upload(path, params['file'])
+    end
+
+    def upload(path, file)
+      directory = Directory.new(path)
+      file_path = "#{path}/#{file[:filename]}"
+
+      ::File.open(file_path, "w") do |f|
+        f.write(file[:tempfile].read)
+      end
+
+      g = Git.open(project_path)
+      g.add(file_path)
+      g.commit("Uploaded #{file[:filename]} via tint")
```

Comment:
This commit can fail, if we upload an overwrite that is identical to the current file.  Should check as in PUT